pkg/object/graph/graph.go

```diff
@@ -101,25 +108,53 @@ func (g *Graph) Size() int {
 	return len(g.edges)
 }
 
-// removeVertex removes the passed vertex as well as any edges
-// into the vertex.
-func (g *Graph) removeVertex(r object.ObjMetadata) {
+// removeVertex removes the passed vertex as well as any edges into the vertex.
+func removeVertex(edges map[object.ObjMetadata]object.ObjMetadataSet, r object.ObjMetadata) {
 	// First, remove the object from all adjacency lists.
-	for v, adj := range g.edges {
-		g.edges[v] = adj.Remove(r)
+	for v, adj := range edges {
+		edges[v] = adj.Remove(r)
 	}
 	// Finally, remove the vertex
-	delete(g.edges, r)
+	delete(edges, r)
 }
 
-// Sort returns the ordered set of vertices after
-// a topological sort.
+// EdgesFrom returns the objects that this object depends on.
+func (g *Graph) EdgesFrom(from object.ObjMetadata) object.ObjMetadataSet {
+	edgesFrom, exists := g.edges[from]
+	if !exists {
+		return nil
+	}
+	c := make(object.ObjMetadataSet, len(edgesFrom))
+	copy(c, edgesFrom)
+	return c
+}
+
+// EdgesTo returns the objects that depend on this object.
+func (g *Graph) EdgesTo(to object.ObjMetadata) object.ObjMetadataSet {
+	edgesTo, exists := g.reverseEdges[to]
+	if !exists {
+		return nil
+	}
+	c := make(object.ObjMetadataSet, len(edgesTo))
+	copy(c, edgesTo)
+	return c
+}
+
+// Sort returns the ordered set of vertices after a topological sort.
```

Comment:
Since the graph now contains`edges` and `reverseEdges`, do we need to update the existing unit test for `Sort`?

Reply:
Yup. It's on the todo list. I just wanted to get feedback on the approach before writing a bunch of tests that i might have to throw away.